src/kernels/jupyter/remoteKernelFinder.web.ts

```diff
@@ -0,0 +1,81 @@
+import { captureTelemetry, sendTelemetryEvent } from '../../telemetry';
+import { getTelemetrySafeLanguage } from '../../telemetry/helpers';
+import { Telemetry } from '../../webviews/webview-side/common/constants';
+import { IRemoteKernelFinder } from '../raw/types';
+import { PreferredRemoteKernelIdProvider } from '../raw/finder/preferredRemoteKernelIdProvider';
+import { getResourceType } from '../../platform/common/utils';
+
+// This is a temporary class to just get the NotebookControllerManager to load in a web context.
+@injectable()
+export class RemoteKernelFinder implements IRemoteKernelFinder {
+    constructor(
+        @inject(PreferredRemoteKernelIdProvider)
+        private readonly preferredRemoteKernelIdProvider: PreferredRemoteKernelIdProvider
+    ) {}
+    @traceDecoratorVerbose('Find remote kernel spec')
+    @captureTelemetry(Telemetry.KernelFinderPerf)
+    @captureTelemetry(Telemetry.KernelListingPerf, { kind: 'remote' })
+    public async findKernel(
+        resource: Resource,
+        connInfo: INotebookProviderConnection | undefined,
+        notebookMetadata?: nbformat.INotebookMetadata,
+        @ignoreLogging() _cancelToken?: CancellationToken
+    ): Promise<KernelConnectionMetadata | undefined> {
+        const resourceType = getResourceType(resource);
+        const telemetrySafeLanguage =
+            resourceType === 'interactive'
+                ? PYTHON_LANGUAGE
+                : getTelemetrySafeLanguage(getLanguageInNotebookMetadata(notebookMetadata) || '');
+        try {
+            // Get list of all of the specs
+            const kernels = await this.listKernels(resource, connInfo);
+
+            // Find the preferred kernel index from the list.
+            const preferred = findPreferredKernel(
+                kernels,
+                resource,
+                notebookMetadata,
+                undefined,
+                this.preferredRemoteKernelIdProvider
+            );
+            sendTelemetryEvent(Telemetry.PreferredKernel, undefined, {
+                result: preferred ? 'found' : 'notfound',
+                resourceType,
+                language: telemetrySafeLanguage
+            });
+            return preferred;
+        } catch (ex) {
+            sendTelemetryEvent(
+                Telemetry.PreferredKernel,
+                undefined,
+                { result: 'failed', resourceType, language: telemetrySafeLanguage },
+                // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                ex as any,
+                true
+            );
+            traceError(`findKernel crashed`, ex);
+        }
+    }
+
+    // Talk to the remote server to determine sessions
+    public async listKernels(
+        resource: Resource,
+        _connInfo: INotebookProviderConnection | undefined
+    ): Promise<KernelConnectionMetadata[]> {
+        sendKernelListTelemetry(resource, []);
+        return [];
```

Comment:
This will likely be the same as the node version once I get all of the session classes below this to work. For now there are two so I can stop this PR at this point.